src/commctl/cli.py

```diff
@@ -528,16 +548,18 @@ def _dispatch(self, client_method):
                 'with a comma.'))
 
         # Prompt for any missing configuration.
-
-        for required in ('username', 'endpoint'):
-            if required not in conf.keys():
-                conf[required] = input(
-                    '{0}: '.format(required.capitalize()))
-
-        # Check password on it's own
-        if 'password' not in conf.keys():
+        if 'endpoint' not in conf.keys():
+            conf['endpoint'] = input('Endpoint: ')
+        if 'kubeconfig' in conf.keys():
+            # If we have a kubeconfig entry move along. It will be handled
+            # once the Client is created.
+            pass
+        elif 'username' not in conf.keys() or 'password' not in conf.keys():
             import getpass
+            print('Missing username/password...')
+            conf['username'] = raw_input('Username: ')
             conf['password'] = getpass.getpass()
```

Comment:
It would not be unusual for a config file to include a username but no password.  In that case we'd be prompting for a username unnecessarily here.

Also, `raw_input` should be `input` given the compatibility hack at the top of `cli.py`, right?